teacherprofile.controller.js

```
@@ -0,0 +1,386 @@
+      name: '',
+      price: ''
+    };
+
+    vm.permissions = {};
+    if (vm.userRole === 'student') {
+      vm.permissions.message = true;
+      vm.permissions.book = true;
+      vm.permissions.unregistered = false;
+    } else if (vm.userRole === 'teacher') {
+      vm.permissions.message = false;
+      vm.permissions.book = false;
+      vm.permissions.unregistered = false;
+    } else {
+      vm.permissions.message = false;
+      vm.permissions.book = false;
+      vm.permissions.unregistered = true;
+    }
+
+    var oneHourDelay = 3600000;
+    var oneMinuteDelay = 60000;
+
+    var id = $stateParams.id;
+    vm.teacherId = id;
+    vm.avatarUrl = Api.url() + '/users/' + id + '/avatars?style=medium';
+
+    vm.getData();
+
+    function getData () {
+      vm.getSubjects();
+      vm.getTeacher();
+    }
+
+    vm.backToPrevious = function () {
+      var cookie = $cookieStore.get('teachers_state_params');
+      var teachers_state_params = JSON.parse(cookie);
+      $state.go('levelup.teachers', teachers_state_params);
+    };
+
+    function getSubjects () {
+      Api.get('/subjects.json')
+        .then(function (response) {
+          vm.allSubjects = response.data.subjects;
+          Api.get('/users/' + id + '/user_subjects.json')
+            .then(function (response) {
+              var subjects = response.data;
+              vm.teacherSubjects = _.sortBy(subjects, 'subject_id');
+              $.each(vm.teacherSubjects, function (index, obj) {
+                for (var i = 0; i < vm.allSubjects.length; i++) {
+                  if (parseInt(vm.allSubjects[i].id) === parseInt(obj.subject_id)) {
+                    obj.name = vm.allSubjects[i].name;
+                    break;
+                  }
+                }
+              });
+              vm.currentSubject = _.select(vm.teacherSubjects, function (sub) { return sub.subject_id === parseInt(vm.currentSubjectId); })[0];
+            })
+            .catch(function (response) {
```

empty catch allows for silent code errors and should be used only if the programmer is 100% aware of its possible consequences - otherwise it's an antipattern in all programming languages